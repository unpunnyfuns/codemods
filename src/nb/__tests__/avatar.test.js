import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('nb/avatar', () => {
  it('migrates Avatar to Nordlys with icon/image object transformations', () => {
    const output = testTransform('nb/avatar', 'nb/avatar', 'tsx', {
      sourceImport: DEFAULT_TEST_OPTIONS.sourceImport,
      targetImport: '@target/components/Avatar',
      tokenImport: DEFAULT_TEST_OPTIONS.tokenImport,
    })
    expect(output).toMatchSnapshot()
  })
})
