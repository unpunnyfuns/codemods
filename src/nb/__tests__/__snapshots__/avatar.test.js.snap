// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/avatar > migrates Avatar to Nordlys with icon/image object transformations 1`] = `
"import { Avatar } from '@target/components/Avatar';
import { View, StyleSheet } from 'react-native';
import { color } from '@design-tokens';

export function MyComponent() {
  return (
    <>

      {/* Dropped during migration (not supported in React Native):
        iconName="user" */
      }
      <View style={styles.avatar0}>
        <Avatar
          size="md"
          icon={{
            name: "user",
            fill: 'blue'
          }} />
      </View>

      {/* Dropped during migration (not supported in React Native):
        imageUri="https://example.com/avatar.jpg"
        isSecondaryColor={true} */
      }
      <Avatar
        size="lg"
        image={{
          source: {
            uri: "https://example.com/avatar.jpg"
          }
        }} />

      {/* Dropped during migration (not supported in React Native):
        imageSource={...}
        placeholder={true} */
      }
      <Avatar
        size="sm"
        image={{
          source: { uri: 'local.jpg' }
        }} />
      <Avatar letters="AB" lettersColor="white" size="md" />
    </>
  );
}

const styles = StyleSheet.create({
  avatar0: {
    backgroundColor: color.blue
  }
});
"
`;
